### PR TITLE
Port get_schema_fields from DRF

### PR DESCRIPTION
**Pull request by rpkilby**

### CODE REVIEW

File: django_filters/rest_framework/backends.py [hunk 1]
```diff
@@ -62,3 +62,10 @@ def to_html(self, request, queryset, view):
         }
         template = loader.get_template(self.template)
         return compat.template_render(template, context)
+
+    def get_fields(self, view):
```
Comment (carltongibson) [inline django_filters/rest_framework/backends.py]: Can you add a Doc string: what's this for? Ta.

Comment (rpkilby) [inline django_filters/rest_framework/backends.py]: The original blame points to core-api/schema generation, but I'm not familiar with that so I don't actually understand how this method is being used. @tomchristie?

Side note - I saw @kevin-brown's comment [here](https://github.com/tomchristie/django-rest-framework/pull/4179#r69496876) and simplified.

File: django_filters/rest_framework/backends.py [hunk 2]
```diff
@@ -62,3 +62,10 @@ def to_html(self, request, queryset, view):
...
+        filter_class = self.get_fitler_class()
```
Comment (rpkilby) [inline django_filters/rest_framework/backends.py]: > get_fitler_class()

And this is why we test anyway.

File: django_filters/rest_framework/backends.py (hunk 1 above)
Comment (tomchristie) [inline django_filters/rest_framework/backends.py]: Indeed, this is used by the schema generation to introspect what fields are required by a filter class. It's not formalized in the docs yet, so we should discuss prior to any 1.0 django-filter / 3.5 rest framework releases.

Comment (rpkilby) [inline django_filters/rest_framework/backends.py]: @tomchristie - could you take a look at 291717f? Does this seem correct? Thanks for the help.

Comment (tomchristie) [inline django_filters/rest_framework/backends.py]: Seems good to me.

**Comment by rpkilby** [comment]
@carltongibson. Anything else that needs to be done here? I'm considering holding off on a doc string until `get_fields` is public/final in DRF's API.

**Comment by tomchristie** [comment]
Okay, these will become public API in 3.5 onwards, as `get_schema_fields`.
Should return a list of `coreapi.Field` instances.

**Comment by carltongibson** [comment]
Awesome. Thanks Tom.

**Comment by rpkilby** [comment]
Hi @carltongibson - this should be ready for review. 

btw, there is technically a flaw with the implementation, but I don't think it's worth fixing. Given the following,
```py
class UserFilter(FilterSet):
    username = filters.CharFilter()
    last_login__between = filters.DateTimeFromToRangeFilter(name='last_login', lookup_expr='range')

    class Meta:
        model = User
```
`get_schema_fields` will incorrectly return query names of `['username', 'last_login__between']` instead of `['username', 'last_login__between_0', 'last_login__between_1']`.

Realistically, this is only an issue for `MultiWidget`, although it could technically apply to any widget. In general, I'm not inclined to try to make this compatible:
- `MultiWidget` isn't really API-friendly to begin with (see #425). 
- There isn't a public API for getting the param names.
- There isn't a trivially reliable way of [introspecting this](https://github.com/django/django/blob/1.10/django/forms/widgets.py#L860-L861).

My thoughts are to either:
- Not worry about it until it becomes relevant. 
- Add a note in the docs that some `MultiWidget`-based filters aren't compatible with all of DRF's features. Instead of using the `*RangeFilter`s, create a CSV-based filter derived from `BaseRangeFilter`. Either that or create, two separate lte/gte filters (see this [thread](https://github.com/philipn/django-rest-framework-filters/issues/125#issuecomment-250366476)).

**Comment by carltongibson** [comment]
OK. This is great. I'll merge this and push a 0.15.3 this morning. 👍

@tomchristie I think we're good from **this side** for DRF 3.5.

**Comment by tomchristie** [comment]
Great stuff!

**Pull request merged**